### ISSUE TITLE
Requirements files out of date: potentially hidden bugs

### ISSUE THREAD

**Issue by pgarrison** [AllenCellModeling/cyto-dl#424]
# Issue
The `requirements.txt` and `pdm.lock` files have not been updated in 5 months, but the `pyproject.toml` has had some revisions. The `requirements.txt` and `pdm.lock` files need to be kept up-to-date so that installation gives the expected dependency versions.

I noticed this issue because [I tried to add support for python 3.11](https://github.com/AllenCellModeling/cyto-dl/pull/420), so I needed to re-run Make Requirments to ensure the locked dependencies support python 3.11. But then I discovered that the tests for currently supported versions 3.9 and 3.10 failed, which suggests there are bigger issues than just the locked dependencies being outdated.

# Potential solution
* Run the Make Requirements action and merge the result, even if it causes tests to fail.
  - File new issues for test failures and work on them immediately
* A github action should automatically update the requirements & lock files whenever `pyproject.toml` changes. This could be done by pushing commits to feature branches when PRs are created/edited, or creating a follow-up PR whenever a relevant PR is merged (see github.com/AllenCell/nuc-morph-analysis for an example)

**Comment by benjijamorris** [comment]
Thanks again for digging through all this! My read is that it's BioIO error and the tests are failing either during image loading or during downstream tests for 3d because the image is a different shape than expected. I can start looking into what the Bioio change might have been, can you think of anyone else who might be able to help spot what package changes made this break?

**Comment by benjijamorris** [comment]
The BioIo versions seem to match what's been working for me locally with python3.10.4 🤔

**Comment by pgarrison** [comment]
Blocked by #427

**Comment by pgarrison** [comment]
It seems there is already a github action to update the lock files, which maybe was just failing before. With #431 created and merged I think this is resolved. Though #433 still needs to be merged.
